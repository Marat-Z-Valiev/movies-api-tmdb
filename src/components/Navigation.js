import React, {useEffect} from "react";
import {Link, NavLink} from "react-router-dom";
import Search from "./Search";
import NavStyled from "./styles/NavStyled";

const Navigation = () => {
	useEffect(() => {
		document.getElementById("app").style.marginTop = `${
			document.querySelector(".menu").offsetHeight + 40
		}px`;
	}, []);

	const stickyStyle = {
		position: "fixed",
		top: "0",
		left: "auto",
		right: "0",
		width: "100%",
	};

	const toggleMenu = () => {
		const liElements = document.querySelectorAll(".item");
		liElements.forEach((element) => {
			element.classList.toggle("open");
		});
		const search = document.querySelector(".search");
		search.classList.toggle("open");
	};

	const closeMenu = () => {
		document.querySelector(".hamburger-button").click();
	};

	const closeMenuAfterClick = () => {
		const liElements = document.querySelectorAll(".item");
		liElements.forEach((element) => {
			element.addEventListener("click", closeMenu);
		});
	};

	useEffect(() => {
		closeMenuAfterClick();
	}, []);

	return (
		<NavStyled>
			<ul className="menu" style={stickyStyle}>
				<li className="logo">
					<Link to={"/"}>
						<h1>Movies</h1>
					</Link>
				</li>
				<li className="item">
					<NavLink to={"/popular"}>Popular</NavLink>
				</li>
				<li className="item">
					<NavLink to="./people">People</NavLink>
				</li>
				<li className="search">
					<Search closeMenu={closeMenu} />
				</li>
				<li className="toggle">
					<button
						type="button"
						className="hamburger-button"
						onClick={toggleMenu}
					>
						<i className="fas fa-bars fa-3x"></i>
					</button>
				</li>
			</ul>
		</NavStyled>
	);
};

export default Navigation;
