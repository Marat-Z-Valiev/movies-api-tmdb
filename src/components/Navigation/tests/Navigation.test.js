import React from "react";
import {shallow, configure} from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import Navigation from "../Navigation";

configure({adapter: new Adapter()});

describe("Navigation", () => {
	it("should render correctly", () => {
		const wrapper = shallow(<Navigation />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});
