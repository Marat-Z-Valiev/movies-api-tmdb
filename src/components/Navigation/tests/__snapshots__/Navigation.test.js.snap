// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render correctly 1`] = `
<NavigationStyled__NavStyled>
  <ul
    className="menu"
    style={
      Object {
        "left": "auto",
        "position": "fixed",
        "right": "0",
        "top": "0",
        "width": "100%",
      }
    }
  >
    <li
      className="logo"
    >
      <Link
        to="/"
      >
        <h1>
          Movies
        </h1>
      </Link>
    </li>
    <li
      className="item"
    >
      <NavLink
        to="/popular"
      >
        Popular
      </NavLink>
    </li>
    <li
      className="item"
    >
      <NavLink
        to="./people"
      >
        People
      </NavLink>
    </li>
    <li
      className="search"
    >
      <withRouter(Search)
        closeMenu={[Function]}
      />
    </li>
    <li
      className="toggle"
    >
      <button
        className="hamburger-button"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fas fa-bars fa-3x"
        />
      </button>
    </li>
  </ul>
</NavigationStyled__NavStyled>
`;
